runtime/runtime/src/ext.rs

```diff
@@ -389,4 +440,44 @@ impl<'a> External for RuntimeExt<'a> {
             .validator_total_stake(self.epoch_id, self.prev_block_hash)
             .map_err(|e| ExternalError::ValidatorError(e).into())
     }
+
+    #[cfg(feature = "protocol_feature_function_call_weight")]
+    fn distribute_unused_gas(&mut self, gas: u64) -> GasDistribution {
```

Comment:
I see that we have the same code here and inside the mock -- could we somehow combine it together ?

Reply:
So the reason I didn't do here is that there is a plan, as mentioned in #6150, to move the logic for receipt handling into VM logic which would remove this duplication. I assume this change would follow this PR. 

The types for action_receipts are currently slightly different and the implementations are in different crates, so creating a temporary abstraction that is exposed publicly doesn't make much sense to me

Reply:
uh, I didn't notice that the action_receipts are different. We should clean it up, but doing it in the next PR is ok.